### ISSUE TITLE
Background location tracking

### ISSUE THREAD

**Issue by OSPFNeighbour** [NSWSESMembers/availability-poc#516]
So we currently cant location track if the app is in the background.

Before I waste hours of my time working on this only to have my hopes and dreams squashed by @sdunster does anyone have a plan on how they want this to work.

Options:

1. custom native code
Pros: Would work very well
Cons: have to write and maintain custom native code for both ios and android

2. jamesisaac-react-native-background-task
Pros: react-native code library
Cons: fastest you can run code is every 15 minutes

3. transistorsoft/react-native-background-geolocation
Pros: react-native code library, external support
Cons: "android use requires a licence", $300 perpetual for 1 app. weirdly its MIT so not sure how they are working that licence fee when you could just fork it and remove the licence requirement.

4. ...

**Comment by sdunster** [comment]
This might be an option or useful for inspiration for the 1st option: https://github.com/mauron85/react-native-background-geolocation

**Comment by sdunster** [comment]
I don't think 1 will be very hard on iOS - I'm sure I've seen react native APIs specifically for this but I'm not sure. I know this is a major buzzkill but I think we can start testing and warn people to leave the app open if they want tracking.

**Comment by sdunster** [comment]
I'm pretty sure it just works™ on iOS, we don't need to do anything except ensure our app is enabled for background updates (pretty sure it already is). https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/LocationAwarenessPG/CoreLocation/CoreLocation.html#//apple_ref/doc/uid/TP40009497-CH2-SW1

I'm sure there's more to it than that though so I will do some more reading.

**Comment by sdunster** [comment]
Ok we probably need to make sure this has been set to true as well. These docs suck, Apple. https://developer.apple.com/documentation/corelocation/cllocationmanager/1620568-allowsbackgroundlocationupdates

**Comment by sdunster** [comment]
we should probably set this too so people are aware that it is using your location/updating and it also provides an easy way for them to get back to it. During testing it will allow us to make sure we turn it off at the appropriate time (when reaching the geofence around HQ for example as well as a fallback timeout). https://developer.apple.com/documentation/corelocation/cllocationmanager/1620568-allowsbackgroundlocationupdates

**Comment by sdunster** [comment]
Looks like all but the status bar in my last comment is implemented on iOS: https://github.com/facebook/react-native/blob/master/Libraries/Geolocation/RCTLocationObserver.m#L215

**Comment by OSPFNeighbour** [comment]
its not only "leave it open" its leave it in the foreground and dont let the phone go to sleep.

**Comment by sdunster** [comment]
https://github.com/corbt/react-native-keep-awake

**Comment by sdunster** [comment]
Hunch: iOS might already work if the app is left in the foreground but the phone is locked anyway.

**Comment by OSPFNeighbour** [comment]
we could use that mauron85/react-native-background-geolocation package and just let it post locations directly to the server and the server do some magic to find which event responses to update

**Comment by OSPFNeighbour** [comment]
Hunch: I dont use ios

**Comment by sdunster** [comment]
yeah longer term this is what I planned to do - we'll have a singleton location responder that activates whenever we're actively responding to an event so we're only ever submitting one location no matter how many things are running/happening etc.

**Comment by OSPFNeighbour** [comment]
how far down the road might this happen. dont want to invest much time into something short time if you are *realistically* going to be coding this any time soon
